torch/csrc/jit/graph_executor.cpp

```diff
@@ -298,6 +298,9 @@ struct GraphExecutorImpl {
   }
 
   static size_t countFlatInputs(const TypePtr& ptr) {
+    if (auto optional_type = ptr->cast<OptionalType>()) {
```

Comment:
I'm not sure about this. Something like Optional[Tuple[int, int]] can be either one or two flat inputs. None, or two ints. What warranted this change exactly ?

Reply:
If you count it as 1 it later doesn't allocate enough space for all the inputs if the input is not none.